tests/unit/components/Editor/ItemEditor.spec.js

```diff
@@ -1,347 +1,541 @@
+        .trigger("click");
+
+      expect(wrapper.vm.localItemDetailList[0].correct_answer).toStrictEqual([
+        0,
+      ]);
+    });
+
+    it("clicking add option adds a new option", async () => {
+      await wrapper.find('[data-test="addOption"]').trigger("click");
+      expect(wrapper.vm.localItemDetailList[0].options.length).toBe(4);
+    });
+
+    it("clicking on delete option deletes option", async () => {
+      await wrapper
+        .findAll('[data-test="option"]')[0]
+        .find('[data-test="endIcon"]')
+        .trigger("click");
+      expect(wrapper.emitted()).toHaveProperty("delete-option");
     });
   });
 
-  it("should render with required values for subjective question", () => {
-    const wrapper = mount(ItemEditor, {
-      props: {
-        itemList: [
-          {
-            type: "question",
-            time: 13,
-          },
-        ],
-        itemDetailList: [
-          {
-            text: "test",
-            type: "subjective",
-          },
-        ],
-        questionTypeIndex: 1,
+  describe("subjective questions", () => {
+    let itemList = [
+      {
+        type: "question",
+        time: 13.3,
+      },
+    ];
+    let itemDetailList = [
+      {
+        text: "test",
+        type: "subjective",
+        has_char_limit: true,
       },
+    ];
+
+    const mountWrapper = (newItemList = null, newItemDetailList = null) => {
+      wrapper = mount(ItemEditor, {
+        props: {
+          itemList: newItemList == null ? itemList : newItemList,
+          itemDetailList:
+            newItemDetailList == null ? itemDetailList : newItemDetailList,
+          questionTypeIndex: 1,
+        },
+      });
+    };
+
+    beforeEach(() => {
+      mountWrapper();
+    });
+
+    it("should render with required values", () => {
+      expect(wrapper.find('[data-test="options"]').exists()).toBeFalsy();
+      expect(
+        wrapper.find('[data-test="subjectiveQuestionContainer"]').exists()
+      ).toBeTruthy();
+    });
+
+    it("enables/disables max char limit", async () => {
+      // checkbox should be checked by default
+      expect(wrapper.vm.localItemDetailList[0].has_char_limit).toBe(true);
+
+      // uncheck
+      await wrapper
+        .find('[data-test="maxCharLimitCheckbox"]')
+        .setChecked(false);
+      expect(wrapper.vm.localItemDetailList[0].has_char_limit).toBe(false);
+
+      await wrapper.find('[data-test="maxCharLimitCheckbox"]').setChecked(true);
+    });
+
+    it("clearing max limit sets it to the minimum value", async () => {
+      await wrapper
+        .find('[data-test="maxCharLimit"]')
+        .find('[data-test="input"]')
+        .setValue("");
+      expect(wrapper.vm.maxCharLimit).toBe(100);
+    });
+
+    it("triggers keyup on max char limit input box", async () => {
+      const maxCharLimitInputKeypress = jest.spyOn(
+        ItemEditor.methods,
+        "maxCharLimitInputKeypress"
+      );
+      mountWrapper();
```

Comment:
We're mounting again here?
Shouldn't we unmount the one first which was already mounted?
And why is there a need of doing this?